### PR TITLE
vRA7 multiple fixes for nested, non-nested and merged parameters for a Blueprint

### LINKED ISSUE #73: Extra Parameters Not Being Sent to vRA
### Versions:
* Version of vmware-vra-gem: 2.6.1
* Version of kitchen-vra gem: 2.6.0
* Version of test-kitchen gem: 1.24.0
* Version of ChefDK: 3.8.14
* Version of Chef Client: 14.10.9
* Version of ruby: 2.5.3p105

### Platform Details

* Version of vRA: 7.5

### Scenario:
I'm trying to pass some extra_parameters to vRA to provision a virtual machine with test kitchen.  We are passing a public key to the virtual machine so the user can ssh to the machine.  The authentication to vRA succeeds and starts to build the machine.  The machine builds successfully but cannot ssh to the box.  From what I see on the vRA/vRO side, the parameters are not getting passed.

### Steps to Reproduce:
Here is a sample of what my kitchen.yml looks like:

driver:
  name: vra
  username: <%= ENV['USERNAME'] %>@domain.com
  password: <%= ENV['MYPASS'] %>
  tenant: dfs
  base_url: https://vra-prod.na.domain.com
  verify_ssl: false
  request_timeout: 1200
  private_key_path: C:\Users\<%= ENV['USERNAME'] %>\.ssh\id_rsa

platforms:
  - name: ChefDK
    driver:
      catalog_id: abcd1234-ab12-ab12-ab12-abcdef123456
      subtenant_id: abcd1234-ab12-ab12-ab12-abcdef123456
      extra_parameters:
        chefdk.ssh_pub_key:
          type: string
          value: 'ssh-rsa abcd1234.......... user@domain'
        rhelVersion:
          type: string
          value: RHEL7

### Expected Result:
```
$ kitchen create
-----> Starting Kitchen (v1.24.0)
Please report a bug if this causes problems.
-----> Creating <default-ChefDK>...
       Building vRA catalog request...
       Catalog request abcd1234-ab12-ab12-ab12-abcdef123456 submitted.
       Current request status: IN_PROGRESS
Waiting For vRA to collect the IP       Server abcd1234-ab12-ab12-ab12-abcdef123456 (hostname) created. Waiting until ready...
       [SSH] Established
       Server abcd1234-ab12-ab12-ab12-abcdef123456 (hostname) ready.
       Finished creating <default-ChefDK> (13m1.60s).
-----> Kitchen is finished. (13m49.70s)
```

### Actual Result:
```
$ kitchen create
-----> Starting Kitchen (v1.24.0)
Please report a bug if this causes problems.
-----> Creating <default-ChefDK>...
       Building vRA catalog request...
       Catalog request abcd1234-ab12-ab12-ab12-abcdef123456 submitted.
       Current request status: IN_PROGRESS
Waiting For vRA to collect the IP       Server abcd1234-ab12-ab12-ab12-abcdef123456 (hostname) created. Waiting until ready...
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
       Waiting for SSH service on 10.10.10.10:22, retrying in 3 seconds
```

### PR DESCRIPTION
**Pull request by stuartpreston**
Signed-off-by: Stuart Preston <stuart@chef.io>

### Description

This PR addresses a few reported issues with the same shared cause; an attempt at implementing the merging of parameters that are specified in a Blueprint with those supplied by another tool (e.g. kitchen-vra Test Kitchen driver) was reverted and left the gem in a state where it would not merge *any* parameters with the ones specified in the Blueprint.

### Issues Resolved

Resolves chef-partners/kitchen-vra/issues/33
Resolves chef-partners/kitchen-vra/issues/30
Resolves #73 

PRs #70 will be closed as this work supercedes that effort.

### Check List

- [x] All tests pass.
- [x] All style checks pass.
- [x] Functionality includes testing.
- [x] Functionality has been documented in the README if applicable

### CODE REVIEW

File: vmware-vra.gemspec [hunk 1]
```diff
@@ -28,6 +28,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "pry",     "~> 0.10"
   spec.add_development_dependency "rake",    "~> 10.0"
   spec.add_development_dependency "rspec",   "~> 3.0"
-  spec.add_development_dependency "webmock", "~> 1.21"
```
Comment (tas50) [inline vmware-vra.gemspec]: I assume you ran into the Ruby 2.6 issues here?

Comment (stuartpreston) [reply vmware-vra.gemspec]: Yes, I bumped Webmock to ~3.5 so that it all now works in Travis.

**Comment by mcascone** [comment]
Forgive me for asking a beginner question here, but what is the best way to install/pull down the new version that fixes this? Will it get pulled on the next chef run? Do i need to do a `chef exec bundle install`?

**Comment by stuartpreston** [comment]
@mcascone It will be in the next release that I'm working on right now... look out for version 2.7.0 on RubyGems later today - you should be able to update this gem with `chef gem install vmware-vra-gem` at that time. Thanks.

**Pull request merged**